addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -53,6 +53,71 @@ def res_partner_enrich_and_create_company(self, partner_id):
 
         return response
 
+    @http.route('/mail_plugin/partner/enrich_and_update_company', type='json', auth='outlook', cors='*')
+    def res_partner_enrich_and_update_company(self, partner_id):
+        """
+        Enriches an existing company using IAP
+        """
+        partner = request.env['res.partner'].browse(partner_id)
+        if not partner.is_company:
+            return {'error': _('Contact must be a company')}
+
+        normalized_email = partner.email_normalized
+        if not normalized_email:
+            return {'error': _('Contact has no valid email')}
+
+        domain = tools.email_domain_extract(normalized_email)
+        iap_data = self._iap_enrich(domain)
+
+        if 'enrichment_info' in iap_data:
```

Comment:
I suppose this indicates an issue. Maybe you could add a small docstring in ``_iap_enrich`` and a small comment here ?

Reply:
Done.
Docstring + comment added